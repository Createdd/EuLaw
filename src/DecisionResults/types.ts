interface DecisionResultsPropsType {
  results: string;
}
interface DecisionResultsStateType {
  isGoing: boolean;
  numberOfGuests: number;
}

export {
  DecisionResultsPropsType,
  DecisionResultsStateType,
};