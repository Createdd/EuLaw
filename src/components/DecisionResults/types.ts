import { TextAnalyticsStateType } from '../TextAnalytics';

interface DecisionResultsPropsType {
  results: TextAnalyticsStateType;
}
interface DecisionResultsStateType {
  //
}

export {
  DecisionResultsPropsType,
  DecisionResultsStateType,
};