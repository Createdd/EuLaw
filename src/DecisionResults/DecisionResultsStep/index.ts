export { default } from './DecisionResultsStep';
export {
    DecisionResultsStepPropsType,
    DecisionResultsStepStateType,
} from './types';
