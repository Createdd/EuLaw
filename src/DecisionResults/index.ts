export { DecisionResults as default } from './DecisionResults';