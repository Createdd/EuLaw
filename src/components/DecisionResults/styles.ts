import styled from 'styled-components';

const StyledForm = styled.form`
  color: blue;
  /* background-color: lightyellow;*/
  > textarea {
    width: 20rem;
    height: 20rem;
    border: 3px solid blue;
  }
`;

export { StyledForm };
