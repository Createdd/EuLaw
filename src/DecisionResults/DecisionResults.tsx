import * as React from 'react';

// import { StyledForm } from './styles';
import { DecisionResultsPropsType, DecisionResultsStateType } from './types';

import DecisionResultsStep from './DecisionResultsStep';

class DecisionResults extends React.Component<
  DecisionResultsPropsType,
  DecisionResultsStateType
> {
  constructor(props: DecisionResultsPropsType) {
    super(props);
    this.state = {
      isGoing: true,
      numberOfGuests: 2
    };
  }

  /* tslint:disable no-any */
  handleInputChange = (event: any) => {
    const target = event.target;
    const value = target.type === 'checkbox' ? target.checked : target.value;
    const name = target.name;

    this.setState({
      [name]: value
    });
  }

  render() {
    return (
      <form>
        <DecisionResultsStep/>
      </form>
    );
  }
}

export { DecisionResults };
