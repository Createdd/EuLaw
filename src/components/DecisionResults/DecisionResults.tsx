import * as React from 'react';
import { intersection as _intersection } from 'lodash';

import { StyledForm } from './styles';
import { DecisionResultsPropsType, DecisionResultsStateType } from './types';

import { TextAnalyticsStateType } from '../TextAnalytics';
import DecisionResultsStep from './DecisionResultsStep';

import MemberStates from '../../data/exampleData/memberStates';

class DecisionResults extends React.Component<
  DecisionResultsPropsType,
  DecisionResultsStateType
> {
  constructor(props: DecisionResultsPropsType) {
    super(props);
  }

  renderDecisionSteps = (results: TextAnalyticsStateType): JSX.Element => {
    if (results.places.length !== 0) {
      const intersect: string[] = _intersection(
        MemberStates.map((str: string) => str.toLowerCase()),
        results.places
      );
      if (intersect.length > 1) {
        return (
          <DecisionResultsStep results={this.props.results} success={true} type="member state"/>
        );
      } else {
        return (
          <DecisionResultsStep results={this.props.results} success={false} type="member state"/>
        );
      }
    }
    return <DecisionResultsStep results={this.props.results} success={false} type="not specified" />;
  }

  render() {
    return (
      <StyledForm>
        <h2>Decision Results</h2>
        <ul>{this.renderDecisionSteps(this.props.results)}</ul>
      </StyledForm>
    );
  }
}

export { DecisionResults };
