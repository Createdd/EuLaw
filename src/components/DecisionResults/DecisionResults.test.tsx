import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { DecisionResults } from './DecisionResults';

const testResults = {
  euTopic: ['test1', 'test2'],
  memberState: ['test1', 'test2']
};

describe('Test DecisionResults', () => {
  it('renders snapshot', () => {
    const tree = renderer
      .create(<DecisionResults results={testResults} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
