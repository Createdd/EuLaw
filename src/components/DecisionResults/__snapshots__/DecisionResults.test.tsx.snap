// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DecisionResults renders snapshot 1`] = `
<form
  className="sc-htpNat jPtAfx"
>
  <h2>
    Decision Results
  </h2>
  <ul>
    <div
      className="sc-bxivhb bUZGRr"
    >
      <li
        className="sc-EHOje jqcrHf"
      >
        <label>
          Is not a 
          member state
           
        </label>
        <input
          checked={false}
          disabled={true}
          name="member state"
          type="checkbox"
        />
      </li>
    </div>
  </ul>
</form>
`;
